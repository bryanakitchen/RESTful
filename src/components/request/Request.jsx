import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TheForm = styled.form`
  padding: 5px;
  border-radius: 5px;
  background: #E9ECEF;
`;

const RadioDiv = styled.div`
  margin-bottom: 10px;
`;

const URLInput = styled.input`
  border-radius: 3px;
  margin-bottom: 10px;
  border: 1px solid gray;
  padding: 3px;
`;

const Labels = styled.label`
background: #e9ecef;
color: #001F54;
  cursor: pointer;
  margin: 4px;
  padding: 3px;
`;

const Radio = styled.input`
  display: none;

  &:checked + ${Labels} {
    background: #034078;
    color: #E9ECEF;
    align-items: center;
    border: 2px solid #101C41;
    border-radius: 3px;
  }
`;

const Request = ({ onChange, onSubmit, url, method, body }) => {
  return (
    <>
      <TheForm onSubmit={onSubmit} >

        <URLInput 
          type="text" 
          placeholder="Enter request URL" 
          name="url" 
          value={url}
          onChange={onChange} />

        <RadioDiv>
          <Radio 
            id="get"
            type="radio" 
            name="method" 
            value="get" 
            checked={method === 'get'}
            onChange={onChange} />
          <Labels htmlFor="get">GET</Labels>

          <Radio 
            id="post"
            type="radio" 
            name="method" 
            value="post"
            checked={method === 'post'}
            onChange={onChange} />
          <Labels htmlFor="post">POST</Labels>

          <Radio 
            id="put"
            type="radio" 
            name="method" 
            value="put"
            onChange={onChange} />
          <Labels htmlFor="put">PUT</Labels>

          <Radio 
            id="delete"
            type="radio" 
            name="method" 
            value="delete"
            checked={method === 'delete'}
            onChange={onChange} />
          <Labels htmlFor="delete">DELETE</Labels>

          <button>Send</button>
        </RadioDiv>

        <textarea 
          placeholder="Raw JSON Body" 
          rows="10" 
          cols="50" 
          name="body" 
          value={body}
          onChange={onChange} ></textarea>
      </TheForm>
    </>
  );
};

Request.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string 
};

export default Request;
