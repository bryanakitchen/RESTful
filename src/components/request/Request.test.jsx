import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Request from './Request';

describe('Request component', () => {
  afterEach(() => cleanup());
  it('renders Request', () => {
    const { asFragment } = render(<Request
      onChange={jest.fn()} 
      onSubmit={jest.fn()} 
      url="www.google.com"
      method="get" 
      body="{}"
    />);    
      
    expect(asFragment()).toMatchSnapshot();
  });
});
