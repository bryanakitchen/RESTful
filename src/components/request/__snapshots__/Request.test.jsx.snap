// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component renders Request 1`] = `
<DocumentFragment>
  <form>
    <input
      name="url"
      placeholder="Enter request URL"
      type="text"
      value="www.google.com"
    />
    <div>
      <input
        checked=""
        name="method"
        type="radio"
        value="get"
      />
      <label
        for="get"
      >
        GET
      </label>
      <input
        name="method"
        type="radio"
        value="post"
      />
      <label
        for="post"
      >
        POST
      </label>
      <input
        name="method"
        type="radio"
        value="put"
      />
      <label
        for="put"
      >
        PUT
      </label>
      <input
        name="method"
        type="radio"
        value="delete"
      />
      <label
        for="delete"
      >
        DELETE
      </label>
      <button>
        Send
      </button>
    </div>
    <textarea
      cols="50"
      name="body"
      placeholder="Raw JSON Body"
      rows="10"
    >
      {}
    </textarea>
  </form>
</DocumentFragment>
`;
